### PR TITLE
Adds endpoint to get distinct values of a json property

### PR DESCRIPTION
**Pull request by JeroenVu**

### CODE REVIEW

File: conf/routes [hunk 1]
```diff
@@ -26,8 +26,10 @@ POST /api/databases/:db/:collection/tx/upsert           controllers.TxController
 
 GET /api/databases/:db/:collection/query                controllers.QueryController.query(db: String, collection: String)
 GET /api/databases/:db/:collection/featurecollection    controllers.QueryController.list(db: String, collection: String)
+GET /api/databases/:db/:collection/distinct             controllers.QueryController.distinct(db: String, collection: String)
 POST /api/databases/:db/:collection/query               controllers.QueryController.query(db: String, collection: String)
 POST /api/databases/:db/:collection/featurecollection   controllers.QueryController.list(db: String, collection: String)
+POST /api/databases/:db/:collection/distinct            controllers.QueryController.distinct(db: String, collection: String)
```
Comment (maesenka) [inline conf/routes:32]: Why a POST in addition to a GET? This is not specified by the RAML spec.

File: app/persistence/postgresql/PostgresqlRepository.scala [hunk 2]
```diff
@@ -602,6 +609,30 @@ class PostgresqlRepository @Inject() (
...
+      val path = {
+        projection.path.path.collect {
+          case KeyPathNode(key) => key
+        }.map(str => s"'$str'").reverse match {
+          case head :: tail => s"->>$head" :: tail.map(str => s"->$str")
+          case Nil => Nil
+        }
+      }.reverse.mkString
+
+      val select = "distinct json" + path
```
Comment (maesenka) [inline app/persistence/postgresql/PostgresqlRepository.scala]: This works only for JSON tables, not for "regular" tables.  Need to check metadata.jsonTable property (see other query methods)

File: conf/routes (hunk 1 above)
Comment (JeroenVu) [reply conf/routes:32]: Analogous to "query" operation to support intersectionGeometryWkt as filter in the request body.

File: app/persistence/postgresql/PostgresqlRepository.scala (hunk 2 above)
Comment (JeroenVu) [reply app/persistence/postgresql/PostgresqlRepository.scala]: See additional commit.

**Pull request merged**